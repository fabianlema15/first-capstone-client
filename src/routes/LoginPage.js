import React from 'react';
import GenericContext from '../contexts/GenericContext'
import LoginContent from '../components/Login/Login'
import GenericApiService from '../services/generic-api-service';
import AuthApiService from '../services/auth-api-service';

class LoginPage extends React.Component{
  static defaultProps = {
    ROUTE : 'logins'
  }

  state = {
    userCode: null
  };

  setUserCode = userCode => {
    this.setState({ userCode })
  }


  setLoginList = objList => {
    this.setState({ objList })
  }

  setError = error => {
    this.setState({ error: error.error })
  }

  clearError = () => {
    this.setState({ error: null })
  }

  checkUserCode = user_code => {
    GenericApiService.getAll(`users/is/${user_code}`)
      .then(result => {
        this.setState({
          userCode: user_code
        })
      })
      .catch(this.setError)
  }

  checkLogin = password => {
    const loginData = {
      user_code: this.state.userCode,
      password: password,
    }
    AuthApiService.login(loginData)
      .then(result => {
        this.props.history.push(`/menu`)
      })
      .catch(this.setError)
  }

  render() {
    const value = {
      clearError: this.clearError,
      userCode: this.state.userCode,
      setUserCode: this.setUserCode,
      error: this.state.error,
      checkUserCode: this.checkUserCode,
      checkLogin: this.checkLogin,
      setError: this.setError
    }

    return (
      <GenericContext.Provider value={value}>
        <LoginContent />
      </GenericContext.Provider>
    )
  }
}

export default LoginPage;
