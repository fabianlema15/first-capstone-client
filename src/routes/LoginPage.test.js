import React from 'react';
import ReactDOM from 'react-dom';
import LoginPage from './LoginPage'
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer';

describe('<LoginPage />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter>
      <LoginPage>
      </LoginPage>
    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Renders LoginPage as expected', () => {
	    const tree = renderer.create(<BrowserRouter>
        <LoginPage>
        </LoginPage>
      </BrowserRouter>).toJSON();
	    expect(tree).toMatchSnapshot();
	  });

})
