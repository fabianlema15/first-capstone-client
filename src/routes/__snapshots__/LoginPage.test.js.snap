// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> Renders LoginPage as expected 1`] = `
<main
  role="main"
>
  <section>
    <fieldset
      className="login-input"
    >
      <fieldset>
        <label
          htmlFor="user_code"
        >
          User Code
          <input
            className="Input "
            id="user_code"
            readOnly={true}
            required={true}
            type="number"
            value=""
          />
        </label>
      </fieldset>
    </fieldset>
  </section>
  <section>
    <div
      className="numbers-panel panel"
    >
      <button
        className="gray"
        onClick={[Function]}
      >
        1
      </button>
      <button
        className="gray"
        onClick={[Function]}
      >
        2
      </button>
      <button
        className="gray"
        onClick={[Function]}
      >
        3
      </button>
      <button
        className="gray"
        onClick={[Function]}
      >
        4
      </button>
      <button
        className="gray"
        onClick={[Function]}
      >
        5
      </button>
      <button
        className="gray"
        onClick={[Function]}
      >
        6
      </button>
      <button
        className="gray"
        onClick={[Function]}
      >
        7
      </button>
      <button
        className="gray"
        onClick={[Function]}
      >
        8
      </button>
      <button
        className="gray"
        onClick={[Function]}
      >
        9
      </button>
      <button
        className="gray"
        onClick={[Function]}
      >
        0
      </button>
    </div>
    <div
      className="buttons"
    >
      <button
        className="orange"
        onClick={[Function]}
      >
        Clear
      </button>
      <button
        className="green"
        onClick={[Function]}
      >
        Next
      </button>
    </div>
  </section>
</main>
`;
